File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:


        output = defaultdict(list)

        for word in strs:

            sorted_word = "".join(sorted(word))

            output[sorted_word].append(word)

        return list(output.values())
        

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        List<List<String>> list = new ArrayList();

        Map<String, List<String>> map = new HashMap();

        for(String word : strs){

            char[] ch = word.toCharArray();
            Arrays.sort(ch);
            String sortedword = Arrays.toString(ch);

            if(!map.containsKey(sortedword)){
                map.put(sortedword,new ArrayList());
            }
            map.get(sortedword).add(word);
            
        }

        return new ArrayList(map.values());
        
    }
}